src/python/sdk/econia_sdk/view/market.py

```diff
@@ -320,6 +320,178 @@ def get_price_levels_all(view: EconiaViewer, market_id: int) -> dict:
+        returns = get_open_orders_paginated(
+            viewer,
+            market_id,
+            min(ask_page_size, max(max_asks - len(asks), 0)),
+            min(bid_page_size, max(max_bids - len(bids), 0)),
+            int(next_ask_idx),
+            int(next_bid_idx),
+            ledger_version,
+        )
+
+        value = returns[0]
+        next_ask_idx = returns[1]
+        next_bid_idx = returns[2]
+
+        for ask in value["asks"]:
+            asks.append(_convert_open_order_value(ask))
+        for bid in value["bids"]:
+            bids.append(_convert_open_order_value(bid))
+
+        if (len(asks) >= max_asks or next_ask_idx == "0") and (
+            len(bids) >= max_bids or next_bid_idx == "0"
+        ):
+            keep_looping = False
+        elif next_ask_idx == "0" or next_bid_idx == "0":
+            if next_ask_idx == "0":
+                ask_page_size = 0
+            else:
+                bid_page_size = 0
+
+    return {"asks": asks, "bids": bids, "market_id": market_id}
+
+
+def get_price_levels_paginated(
+    viewer: EconiaViewer,
+    market_id: int,
+    ask_page_size: int,
+    bid_page_size: int,
+    next_ask_idx: int,
+    next_bid_idx: int,
+    ledger_version: int,
+):
+    returns = viewer.get_returns(
+        "market",
+        "get_price_levels_paginated",
+        [],
+        [
+            str(market_id),
+            str(ask_page_size),
+            str(bid_page_size),
+            str(next_ask_idx),
+            str(next_bid_idx),
+        ],
+        ledger_version=ledger_version,
+    )
+    return returns
+
+
+def get_price_levels_with_pagination(
+    viewer: EconiaViewer,
+    market_id: int,
+    max_asks: int = 100,
+    max_bids: int = 100,
+    page_size: int = 100,
+) -> dict:
+    info = viewer.aptos_client.info()
+    ledger_version = int(info["ledger_version"])
+
+    next_ask_idx = "0"
+    next_bid_idx = "0"
+    bid_page_size = ask_page_size = page_size
+
+    asks = dict()
+    asks_size = 0
+    bids = dict()
+    bids_size = 0
+
+    keep_looping = True
+    while keep_looping:
+        returns = get_price_levels_paginated(
+            viewer,
+            market_id,
+            ask_page_size,
+            bid_page_size,
+            int(next_ask_idx),
+            int(next_bid_idx),
+            ledger_version,
+        )
+
+        value = returns[0]
+        next_ask_idx = returns[1]
+        next_bid_idx = returns[2]
+
+        for ask in value["asks"]:
+            price = int(ask["price"])
+            size = int(ask["size"])
+            if price in asks:
+                asks[price] += size
+            else:
+                asks[price] = size
+        for bid in value["bids"]:
+            price = int(bid["price"])
+            size = int(bid["size"])
+            if price in bids:
+                bids[price] += size
+            else:
+                bids[price] = size
+        asks_size += ask_page_size
+        bids_size += bid_page_size
```

Comment:
This is not necessarily true, because even if you ask for a page of size 100 you might only get 50 orders back

So how about incrementing `asks_size` / `bids_size` by the length of the return asks/bids?

Reply:
Also, how about naming this `n_indexed_asks` and `n_indexed_bids`?

Reply:
In this case we can't, because the length of the price levels that we get back is not necessarily equal to the length of the orders it covers. This is basically a way of knowing for sure whether or not we could possibly be over the limit.

Is there a case where we will break before getting all necessary items? I don't think so.

Reply:
> the length of the price levels that we get back is not necessarily equal to the length of the orders it covers

Ahh, right, good point

> This is basically a way of knowing for sure whether or not we could possibly be over the limit.

Indeed, good call

> Is there a case where we will break before getting all necessary items? I don't think so.

Only if indexing is done, but if that's the case next idx will be 0 and will trigger the break condition anyways